src/common/ant-registry.ts

```diff
@@ -70,6 +77,41 @@ export class AoANTRegistryReadable implements AoANTRegistryRead {
     } else {
       throw new InvalidContractConfigurationError();
     }
+
+    if (config?.hyperbeamUrl !== undefined) {
+      this.hyperbeamUrl = new URL(config.hyperbeamUrl).toString();
+      this.checkHyperBeamPromise = this.checkHyperBeamCompatibility();
+    }
+  }
+
+  /**
+   * Check if the process is HyperBeam compatible. If so, we'll use the HyperBeam node to fetch the state.
+   *
+   * @returns {Promise<boolean>} True if the process is HyperBeam compatible, false otherwise.
+   */
+  private async checkHyperBeamCompatibility(): Promise<boolean> {
+    if (this.hyperbeamUrl === undefined) {
+      return Promise.resolve(false);
+    }
+    if (this.checkHyperBeamPromise !== undefined) {
+      return this.checkHyperBeamPromise;
+    }
+    console.debug('Checking HyperBeam compatibility');
+    this.checkHyperBeamPromise = fetch(
+      `${this.hyperbeamUrl.toString()}${this.process.processId}~process@1.0/now/acl`,
```

Comment:
```suggestion
      `${this.hyperbeamUrl.toString()}${this.process.processId}~process@1.0/now/acl`,
```
per Tom's rec, shouldn't we be using `/now/cache/acl`

Reply:
https://github.com/ar-io/ar-io-sdk/pull/508/commits/873d356739797f861d2d5bd3c042888f52c4b995